UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/MSIBannerNotificationService.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Ganss.XSS;
+using Microsoft.Extensions.Options;
+using UKHO.MaritimeSafetyInformation.Common.Configuration;
+using UKHO.MaritimeSafetyInformation.Common.Helpers;
+using UKHO.MaritimeSafetyInformation.Common.Models.AzureTableEntities;
+using UKHO.MaritimeSafetyInformation.Web.Services.Interfaces;
+
+namespace UKHO.MaritimeSafetyInformation.Web.Services
+{
+    public class MSIBannerNotificationService : IMSIBannerNotificationService
+    {
+        private readonly IOptions<CacheConfiguration> _cacheConfiguration;
+        private readonly IAzureStorageService _azureStorageService;
+        private readonly IAzureTableStorageClient _azureTableStorageClient;
+        private readonly IOptions<BannerNotificationConfiguration> _bannerNotificationConfiguration;
+        
+        private string ConnectionString => _azureStorageService.GetStorageAccountConnectionString(_cacheConfiguration.Value.CacheStorageAccountName, _cacheConfiguration.Value.CacheStorageAccountKey);
+
+        public MSIBannerNotificationService(IOptions<CacheConfiguration> cacheConfiguration, IAzureStorageService azureStorageService, IAzureTableStorageClient azureTableStorageClient, IOptions<BannerNotificationConfiguration> bannerNotificationConfiguration)
+        {
+            _cacheConfiguration = cacheConfiguration;
+            _azureStorageService = azureStorageService;
+            _azureTableStorageClient = azureTableStorageClient;
+            _bannerNotificationConfiguration = bannerNotificationConfiguration; 
+        }
+
+        public async Task<string> GetBannerNotification()
+        {
+            string bannerNotificationMessage = null;
+
+            if (_bannerNotificationConfiguration.Value.IsBannerNotificationEnabled)
+            {
+                MsiBannerNotificationEntity msiBannerNotificationEntity = await _azureTableStorageClient.GetSingleEntityAsync(_cacheConfiguration.Value.MsiBannerNotificationTableName, ConnectionString);
+                if (msiBannerNotificationEntity != null)
+                {
+                    HtmlSanitizer sanitizer = new ();
+                    bannerNotificationMessage = sanitizer.Sanitize(msiBannerNotificationEntity.Message);
```

Comment:
Please add test case which proves XSS details are removed from banner message. 
You can have test s given below:

[Test]
        public async Task WhenBannerNotificationFoundWithXSS_ThenValueExcludingXSSForBannerNotificationMessageIsAssigned()
        {
            string message = @"This system will be under maintenance. For more details <a href=""https://www.test.com"" target=""_blank"">click here</a><script>alert('XSS attack!')</script>";

            string expectedMessage = @"This system will be under maintenance. For more details <a href=""https://www.test.com"" target=""_blank"">click here</a>";

            A.CallTo(() => _fakeAzureTableStorageClient.GetSingleEntityAsync(A<string>.Ignored, A<string>.Ignored)).Returns(new MsiBannerNotificationEntity() { Message = message });

            string result = await _mSIBannerNotificationService.GetBannerNotification();

            Assert.AreEqual(expectedMessage, result);
        }

Reply:
done required changes.